pkg/trace/sampler/signature.go

```diff
@@ -71,23 +71,22 @@ func (s ServiceSignature) String() string {
 	return "service:" + s.Name + ",env:" + s.Env
 }
 
-func computeSpanHash(span *pb.Span, env string) spanHash {
+func computeSpanHash(span *pb.Span, env string, withResource bool) spanHash {
 	h := fnv.New32a()
 	h.Write([]byte(env))
 	h.Write([]byte(span.Service))
 	h.Write([]byte(span.Name))
 	h.Write([]byte{byte(span.Error)})
-
-	return spanHash(h.Sum32())
-}
-
-func computeRootHash(span pb.Span, env string) spanHash {
-	h := fnv.New32a()
-	h.Write([]byte(env))
-	h.Write([]byte(span.Service))
-	h.Write([]byte(span.Name))
-	h.Write([]byte(span.Resource))
-	h.Write([]byte{byte(span.Error)})
-
+	if withResource {
+		h.Write([]byte(span.Resource))
+	}
+	statusCode, ok := getMeta(span, KeyHTTPStatusCode)
+	if ok {
```

Comment:
I think shorter is sweeter...
```suggestion
	if code, ok := getMeta(span, KeyHTTPStatusCode); ok {
```